src/zenml/integrations/langchain/materializers/openai_embedding_materializer.py

```diff
@@ -0,0 +1,59 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the langchain openai embedding materializer."""
+
+
+import os
+import pickle
+from typing import Type
+
+from langchain.embeddings import OpenAIEmbeddings
+
+from zenml.enums import ArtifactType
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+DEFAULT_FILENAME = "embedding.pkl"
+
+
+class LangchainOpenaiEmbeddingMaterializer(BaseMaterializer):
+
+    ASSOCIATED_ARTIFACT_TYPE = ArtifactType.MODEL
+    ASSOCIATED_TYPES = (OpenAIEmbeddings,)
+
+    def load(self, data_type: Type[OpenAIEmbeddings]) -> OpenAIEmbeddings:
+        """Reads a openai embedding from a pickle file.
+
+        Args:
+            data_type: The type of the vector store.
+
+        Returns:
+            The vector store.
+        """
+        super().load(data_type)
+        filepath = os.path.join(self.uri, DEFAULT_FILENAME)
+        with fileio.open(filepath, "rb") as fid:
+            embedding = pickle.load(fid)
```

Comment:
I'd need some extra input / support on implementing the checks for this. AFAIK the pickle file contains no metadata to allow you to determine what Python version was used to save it. Are you suggesting that instead of pickling just the vector store, we pickle an object which contains the metadata and the object, e.g.:

```python
for_plcking = {"vector_store": VectorStore, "python_version": "3.8"}
```

Or what is the proposal here? I looked at the documentation on metadata for artifacts but it didn't seem to apply here. The metadata we extract for artifacts all seems to be extracted from the object itself.

Reply:
As discussed yesterday, I've now added in validation for the pickle files.